### DISCUSSION TITLE
Scrolling/moving easily between frames

### DISCUSSION THREAD

**Discussion by smistad** [smistad/annotationweb#50] [Ideas]
It has been suggested to use the scroll wheel to move easily back and forth between frames.
An option is also to use the arrow keys, left and right.
Another idea is to use the CTRL/SHIFT/ALT keys + left/right/mousewheel to move between annotation frames.

@androst I believe you implemented a zoom functionality once. Is this a useful feature? Quite rare that you need to zoom in ultrasound? How can we enable both zoom and frame scrolling?

Any ideas are welcome. This feature is anyway easy to implement, we just need to decide what is best.

**Comment by jvoor** [comment]
Zoom could be mapped to CTRL+mouse wheel - similar to most Microsoft applications? Frame cycling would be used more often I think.

**Reply by smistad** [reply]
In Firefox CTRL+mousewheel performs zoom in the entire browser. Not sure if possible or wise to override this.
I think now in annotationweb, using mousewheel when having pointer over image, zooms the image. But there is no way to pan though..

**Reply by jvoor** [reply]
ahh... maybe using up/down arrow for zoom then? right and left arrows for skipping to next/previous annotation frames?

**Reply by androst** [reply]
Useful features and suggestions. I also got two similar/related requests regarding scrolling frames automatically around the frame being annotated (e.g. 2-5 frames) to improve the detection of borders from speckle motion. Perhaps that could be a button as well?

Shift + Mouse scroll for zoom the canvas? As you say, zoom is probably not used that often, and even less if the canvas is bigger (as it probably will be in the future). I agree that arrows should also be mapped to moving the frames.

**Reply by jvoor** [reply]
Would scrolling the mouse wheel back and forth not accomplish the same thing (as automatically playing adjacent frames)? One problem I foresee is that the spline is linked to the annotation frame, so when you move frames back and forth the spline will appear and disappear. Probably not a big issue though.

**Reply by androst** [reply]
Yes, that would be the same thing. I meant it as an additional feature, e.g. having an enable/disable button to automatically perform this operation while annotating.

Agree, the spline appearing/disappearing might be an issue. In some software they gradually blend it with distance from the annotation frame, but would probably require some time to implement if relevant here.

**Reply by jvoor** [reply]
okay got it. Good idea with the "time blending" of the spline but I can imagine it would involve a lot of work and it is probably not going to be so important.
